corehq/apps/hqadmin/tasks.py

```diff
@@ -201,3 +204,39 @@ def track_pg_limits():
                 cursor.execute(f'select last_value from "{sequence}"')
                 current_value = cursor.fetchone()[0]
                 metrics_gauge('commcare.postgres.sequence.current_value', current_value, {'table': table, 'database': db})
+
+
+@periodic_task(queue='background_queue', run_every=crontab(minute="0", hour="4"))
+def reconcile_es_cases():
+    _reconcile_es_data('case', 'commcare.elasticsearch.stale_cases', 'reconcile_es_cases')
+
+
+@periodic_task(queue='background_queue', run_every=crontab(minute="0", hour="4"))
+def reconcile_es_forms():
+    _reconcile_es_data('form', 'commcare.elasticsearch.stale_forms', 'reconcile_es_forms')
+
+
+def _reconcile_es_data(data_type, metric, blob_parent_id):
+    today = date.today()
+    two_days_ago = today - timedelta(days=2)
```

Comment:
You are correct that this is a date not a datetime. That being said, the sweep window does change a bit based on when it runs. The window is anything after the start date, which means we will never miss a doc, but it does in theory include docs that will also be in the next two sweeps. I did that because I did not want to get into a situation where a "frozen" doc in ES kept getting updated in postgres and so missed all the windows. For example a case updated daily but frozen in ES, could always miss the window if we had a firm end date.